### PR TITLE
Changed device registration to use email and AuthKey + other small changes

### PR DESCRIPTION
**Pull request by klew**
Changed device registration to use email and AuthKey (instead of location id and password). Moved localIp handling to Network class. Fixed bug with DS18B20 wrong status after lost connection with server. Changed library metafile, so it won't generate warning for EPS8266. 
Changed library version to 2.3.0

### CODE REVIEW

File: libraries/SuplaDevice/examples/SuplaSomfy/SuplaSomfy.ino [hunk 1]
```diff
@@ -128,9 +128,11 @@ void supla_Timer() {
 void setup() {
   Serial.begin(115200);
 
-  // ﻿Replace the falowing GUID
+  // Replace the falowing GUID
   char GUID[SUPLA_GUID_SIZE] = {0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00};
-  // ﻿with GUID that you can retrieve from https://www.supla.org/arduino/get-guid
+  // with GUID that you can retrieve from https://www.supla.org/arduino/get-guid
```
Comment (przemyslawzygmunt) [inline libraries/SuplaDevice/examples/SuplaSomfy/SuplaSomfy.ino:133]: I have created a dedicated url
https://www.supla.org/pl/arduino/get-authkey

File: libraries/SuplaDevice/examples/Supla_DHT/Supla_DHT.ino [hunk 2]
```diff
@@ -69,10 +69,11 @@ void setup() {
   // Set temperature/humidity callback
   SuplaDevice.setTemperatureHumidityCallback(&get_temperature_and_humidity);
 
-  // ﻿Replace the falowing GUID
+  // Replace the falowing GUID
   char GUID[SUPLA_GUID_SIZE] = {0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00};
-  // ﻿with GUID that you can retrieve from https://www.supla.org/arduino/get-guid
-
+  // with GUID that you can retrieve from https://www.supla.org/arduino/get-guid
```
Comment (przemyslawzygmunt) [inline libraries/SuplaDevice/examples/Supla_DHT/Supla_DHT.ino:74]: I have created a dedicated url
https://www.supla.org/pl/arduino/get-authkey

File: libraries/SuplaDevice/examples/Supla_DallasTemperature/Supla_DallasTemperature.ino [hunk 3]
```diff
@@ -66,22 +65,23 @@ void setup() {
...
   SuplaDevice.setTemperatureCallback(&get_temperature);
- 
-  // ﻿Replace the falowing GUID
-  char GUID[SUPLA_GUID_SIZE] = {0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00};
-  // ﻿with GUID that you can retrieve from https://www.supla.org/arduino/get-guid
 
+  // Replace the falowing GUID
+  char GUID[SUPLA_GUID_SIZE] = {0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00};
+  // with GUID that you can retrieve from https://www.supla.org/arduino/get-guid
+  // Use the same link as for GUID and generate AuthKey:
```
Comment (przemyslawzygmunt) [inline libraries/SuplaDevice/examples/Supla_DallasTemperature/Supla_DallasTemperature.ino:75]: I have created a dedicated url
https://www.supla.org/pl/arduino/get-authkey

File: libraries/SuplaDevice/examples/Supla_RGBW/Supla_RGBW.ino [hunk 4]
```diff
@@ -92,10 +92,11 @@ void setup() {
   // Set RGBW callbacks
   SuplaDevice.setRGBWCallbacks(&get_rgbw_value, &set_rgbw_value);
 
-  // ﻿Replace the falowing GUID
+  // Replace the falowing GUID
   char GUID[SUPLA_GUID_SIZE] = {0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00};
-  // ﻿with GUID that you can retrieve from https://www.supla.org/arduino/get-guid
-
+  // with GUID that you can retrieve from https://www.supla.org/arduino/get-guid
+  // Use the same link as for GUID and generate AuthKey:
```
Comment (przemyslawzygmunt) [inline libraries/SuplaDevice/examples/Supla_RGBW/Supla_RGBW.ino:98]: I have created a dedicated url
https://www.supla.org/pl/arduino/get-authkey

File: libraries/SuplaDevice/examples/Supla_RollerShutter/Supla_RollerShutter.ino [hunk 5]
```diff
@@ -54,11 +54,13 @@ void supla_rs_LoadSettings(int channelNumber, unsigned int *full_opening_time, u
...
   Serial.begin(9600);
- 
-  // ﻿Replace the falowing GUID
-  char GUID[SUPLA_GUID_SIZE] = {0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00};
-  // ﻿with GUID that you can retrieve from https://www.supla.org/arduino/get-guid
 
+  // Replace the falowing GUID
+  char GUID[SUPLA_GUID_SIZE] = {0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00};
+  // with GUID that you can retrieve from https://www.supla.org/arduino/get-guid
+  // Use the same link as for GUID and generate AuthKey:
```
Comment (przemyslawzygmunt) [inline libraries/SuplaDevice/examples/Supla_RollerShutter/Supla_RollerShutter.ino:61]: I have created a dedicated url
https://www.supla.org/pl/arduino/get-authkey

File: libraries/SuplaDevice/examples/Supla_RollerShutter_FRAM/Supla_RollerShutter_FRAM.ino [hunk 6]
```diff
@@ -130,10 +130,11 @@ void setup() {
     Serial.println("No SPI FRAM found ... check your connections\r\n");
   }
  
-  // ﻿Replace the falowing GUID
+  // Replace the falowing GUID
   char GUID[SUPLA_GUID_SIZE] = {0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00};
-  // ﻿with GUID that you can retrieve from https://www.supla.org/arduino/get-guid
-
+  // with GUID that you can retrieve from https://www.supla.org/arduino/get-guid
+  // Use the same link as for GUID and generate AuthKey:
```
Comment (przemyslawzygmunt) [inline libraries/SuplaDevice/examples/Supla_RollerShutter_FRAM/Supla_RollerShutter_FRAM.ino:136]: I have created a dedicated url
https://www.supla.org/pl/arduino/get-authkey

File: readme.md [hunk 7]
```diff
@@ -42,9 +42,100 @@ Each example can run on Arduino Mega or ESP8266 board. Please read comments in e
...
+  SuplaDevice.begin(GUID,              // Global Unique Identifier 
+                    "svr1.supla.org",  // SUPLA server address
+                    "mail@address.pl",     // Email address
+                    AUTHKEY);          // Authentication key
+```
+What is different? Well, GUID and Supla server address it the same as previously. MAC address, location ID, location password are removed.
+MAC address was moved to network interface class. Location ID and password were replaced with new authentication method - via email address
+and authentication key. You can generate your authentication key in the same way as GUID (it is actually in exactly the same format):
+```
+// Generate AUTHKEY from https://www.supla.org/arduino/get-guid
```
Comment (przemyslawzygmunt) [inline readme.md:111]: I have created a dedicated url
https://www.supla.org/pl/arduino/get-authkey

**Review by przemyslawzygmunt — CHANGES_REQUESTED** [review]

File: libraries/SuplaDevice/examples/SuplaSomfy/SuplaSomfy.ino (hunk 1 above)
Comment (klew) [reply libraries/SuplaDevice/examples/SuplaSomfy/SuplaSomfy.ino:133]: This link is to pl version. "get-guid" is without /pl/ and it works. "get-authkey" doesn't work without locale. Could you please create link that will be not assigned to specific language?

Comment (przemyslawzygmunt) [reply libraries/SuplaDevice/examples/SuplaSomfy/SuplaSomfy.ino:133]: Done

**Pull request merged**